src/DynamicData/List/Internal/MergeManyCacheChangeSets.cs

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) 2011-2023 Roland Pheasant. All rights reserved.
+// Roland Pheasant licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for full license information.
+
+using System.Reactive.Disposables;
+using System.Reactive.Linq;
+using DynamicData.Cache.Internal;
+
+namespace DynamicData.List.Internal;
+
+/// <summary>
+/// Operator that is similiar to MergeMany but intelligently handles Cache ChangeSets.
+/// </summary>
+internal sealed class MergeManyCacheChangeSets<TObject, TDestination, TDestinationKey>
+    where TObject : notnull
+    where TDestination : notnull
+    where TDestinationKey : notnull
+{
+    private readonly IObservable<IChangeSet<TObject>> _source;
+
+    private readonly Func<TObject, IObservable<IChangeSet<TDestination, TDestinationKey>>> _changeSetSelector;
+
+    private readonly IComparer<TDestination>? _comparer;
+
+    private readonly IEqualityComparer<TDestination>? _equalityComparer;
+
+    public MergeManyCacheChangeSets(IObservable<IChangeSet<TObject>> source, Func<TObject, IObservable<IChangeSet<TDestination, TDestinationKey>>> selector, IEqualityComparer<TDestination>? equalityComparer, IComparer<TDestination>? comparer)
+    {
+        _source = source;
+        _changeSetSelector = selector;
+        _comparer = comparer;
+        _equalityComparer = equalityComparer;
+    }
+
+    public IObservable<IChangeSet<TDestination, TDestinationKey>> Run()
+    {
+        return Observable.Create<IChangeSet<TDestination, TDestinationKey>>(
+            observer =>
+            {
+                var locker = new object();
+
+                // Transform to an observable list of merge containers.
+                var sourceListOfCaches = _source
+                                            .Transform(obj => new ChangeSetCache<TDestination, TDestinationKey>(_changeSetSelector(obj)))
+                                            .Synchronize(locker)
+                                            .AsObservableList();
+
+                var shared = sourceListOfCaches.Connect().Publish();
+
+                // this is manages all of the changes
+                var changeTracker = new ChangeSetMergeTracker<TDestination, TDestinationKey>(() => sourceListOfCaches.Items.ToArray(), _comparer, _equalityComparer);
+
+                // when a source item is removed, all of its sub-items need to be removed
+                var removedItems = shared
+                    .OnItemRemoved(mc => changeTracker.RemoveItems(mc.Cache.KeyValues, observer))
+                    .Subscribe();
```

Comment:
`.Subscribe()` calls here should probably all be `.SubscribeSafe()` as its documentation recommends its usage within operators. This literally just `try`/`catch`es the `.Subscribe()` call and feeds anything it catches to `onError`. This is important for operators, because it prevents things like `.OnNext()` or `.OnCompleted()` from throwing, when operators that orchestrate different streams together (like, say `.Concat()`) are involved. 

I was also going to suggest passing `observer.OnError` to this subscription, but I'm less sold on that, after thinking it through. Since the source here is shared with `.Publish()` the only errors that can occur on this subscription (that aren't already going to be propagated from the "main" subscription) are from `.OnItemRemoved()`, which doesn't involve any user code. If there's a bug on our part, that's going to throw out of `.Subscribe()` or `.OnNext()` or `.OnComplete()` or whatever the user was doing that triggered the removal operation. But maybe that's actually preferable, since it's not the user's fault?

Reply:
Since it's a shared Observable, I don't think they all need to be SubscribeSafe.  This one in particular is only used internally, so if it throws something is really wrong.

Maybe the one below should be SubscribeSafe so that the consumer can detect a bad subscription via their OnError.  That one also passes in `observer.OnError`.

But I do like Roland's suggestion of uniformly addressing these as a single PR.